addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/CoffeeSystemChannelSelector.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.miele.handler;
+
+import java.lang.reflect.Method;
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.miele.handler.MieleBridgeHandler.DeviceMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+
+/**
+ * The {@link ApplianceChannelSelector} for coffee systems
+ *
+ * @author Karel Goderis - Initial contribution
+ * @author Kai Kreuzer - Changed START_TIME to DateTimeType
+ * @author maplesteve - Adoption for the coffeemaker
+ */
+public enum CoffeeSystemChannelSelector implements ApplianceChannelSelector {
+
+    PRODUCT_TYPE("productTypeId", "productType", StringType.class, true),
+    DEVICE_TYPE("mieleDeviceType", "deviceType", StringType.class, true),
+    BRAND_ID("brandId", "brandId", StringType.class, true),
+    COMPANY_ID("companyId", "companyId", StringType.class, true),
+    STATE("state", "state", StringType.class, false),
+    PROGRAMID("programId", "program", StringType.class, false),
+    PROGRAMTYPE("programType", "type", StringType.class, false),
+    PROGRAMPHASE("phase", "phase", StringType.class, false),
+    // lightingStatus signalFailure signalInfo
+    DOOR("signalDoor", "door", OpenClosedType.class, false) {
+        @Override
+
+        public State getState(String s, DeviceMetaData dmd) {
+            if (s.equals("true")) {
```

Comment:
O.k. Since I more or less copied the code from the other ...ChannelSelector classes, this issue affects them also. I'll fix it in the other classes as well.